### ISSUE TITLE
Add golden test review process to release cycle

### ISSUE THREAD

**Issue by gaaclarke** [flutter/flutter#144835]
Currently, when we create release builds of the engine, the golden image test results are not checked (automated or manually).  Some of them are not even executed, only certain invocations of "run_tests.py" are used.

We'd like to have confidence that our cherry-picks are not causing some failure, so ideally the golden tests would be executed and checked.

The golden image results can not be naively uploaded to skia gold because skia gold has a consistent timeline that checks against one branch.  Comparing a golden from a release branch to the main branch would confuse skia gold and lead to situations where there is no correct course forward.

A potential solution would be to create new tests related to the release branch.  For example `aiks_foo_test.png` could become `aiks_foo_test.png.v12_34_5` for release `12.32.5`.  The tests would first be uploaded with zero cherry-picks and blanket approved.  Those images will then serve as the basis from which cherry-pick golden results will be compared.

cc @zanderso @matanlurey

**Comment by chinmaygarde** [comment]
From the Impeller Weekly: The next steps were to append the release name (branch?) to the test to create a new variant for each release.

**Comment by matanlurey** [comment]
The proposal has steeped for a few weeks and doesn't seem to have strong objection, so I'll start working on this.

**Comment by matanlurey** [comment]
Filed https://github.com/flutter/flutter/issues/146078 to ask for infra support since I [did not get a response on Discord](https://discord.com/channels/608014603317936148/608021351567065092/1223342704294105151).

**Comment by matanlurey** [comment]
@gaaclarke Is this effectively closed?

**Comment by gaaclarke** [comment]
Yes, we have a way that we can do golden test reviews, we've done a trial run to prove that it works.  The only outstanding work may be for any documentation surrounding releases to get updated, if they exist.  It may be worth mentioning to Chris in case he knows of anything that should be updated.

**Comment by matanlurey** [comment]
I hesitate to add a lot of extra things on Chris' place right now, but at the same time this is pretty important for us to land Impeller and keep the quality high. @christopherfujino or @itsjustkevin could we help update the release documentation, if any?

Quick summary for the uninitiated: https://github.com/flutter/engine/tree/main/testing/skia_gold_client#release-testing

**Comment by gaaclarke** [comment]
Or just point us to the documentation and we can update it.
